tests/fixtures/optimizer/predicate_pushdown.sql

```diff
@@ -0,0 +1,11 @@
+SELECT x.a AS a FROM (SELECT a FROM x AS x) AS x JOIN y WHERE x.a = 1 AND x.b = 1 AND y.a = 1;
+SELECT x.a AS a FROM (SELECT a FROM x AS x WHERE a = 1 AND x.b = 1) AS x JOIN y ON y.a = 1 WHERE TRUE AND TRUE AND TRUE;
```

Comment:
Not relevant to this PR, but do we need to run `simplify` again? And maybe pop off the entire WHERE clause if its just true?

Reply:
Also, it's strange the predicate became `a = 1`, unqualified. Is that right?

Reply:
correct, this is only testing predicate_pushdown, if you run it through full optimizer, all these things will happen

these tests isolate to one optimizer function

Reply:
`qualify_columns` happens before `predicate_pushdown` right now.

Reply:
(..so I don't think we'd qualify `a` with the full optimizer yet)